internal/http/client.go

```diff
@@ -0,0 +1,77 @@
+package http
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net/http"
+	"os"
+)
+
+type Client struct {
+	client *http.Client
+}
+
+func NewHTTPClient() *Client {
+
+	tlsConfig, err := CreateTLSClientConfig()
+	if err != nil {
+		return nil
+	}
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsConfig,
+		},
+	}
+
+	return &Client{
+		client: client,
+	}
+}
+
+func CreateTLSClientConfig() (*tls.Config, error) {
+	var certData, keyData, rootCA []byte
+	var err error
+
+	// TODO: get files from conf
+	certData, err = os.ReadFile("/etc/pki/consumer/cert.pem")
+	if err != nil {
+		return nil, fmt.Errorf("cannot read cert-file: %w", err)
+	}
+
+	keyData, err = os.ReadFile("/etc/pki/consumer/key.pem")
+	if err != nil {
+		return nil, fmt.Errorf("cannot read key-file: %w", err)
+	}
+
+	rootCA, err = os.ReadFile("/etc/rhsm/ca/redhat-uep.pem")
+	if err != nil {
+		return nil, fmt.Errorf("cannot read ca-file: %w", err)
+	}
+
+	tlsConfig := &tls.Config{}
+	cert, err := tls.X509KeyPair(certData, keyData)
+	if err != nil {
+		return nil, fmt.Errorf("cannot create key pair: %w", err)
+	}
+	tlsConfig.Certificates = []tls.Certificate{cert}
+
+	pool, err := x509.SystemCertPool()
+	if err != nil {
+		return nil, fmt.Errorf("cannot copy system certificate pool: %w", err)
+	}
+
+	pool.AppendCertsFromPEM(rootCA)
+
+	return tlsConfig, nil
+}
+
+func (c *Client) Get(url string) (*http.Response, error) {
```

Comment:
It's customary to define methods on a struct closer to the struct declaration. Move this above `CreateTLSClientConfig`, or move the declaration of `Client` to below the top-level package functions.